src/Build/Microsoft.Build.csproj

```diff
@@ -122,6 +122,10 @@
     <Compile Include="..\Shared\NodeShutdown.cs" />
     <Compile Include="..\Shared\NamedPipeUtil.cs" />
     <Compile Include="..\Shared\ProcessExtensions.cs" />
+    <Compile Include="..\Shared\PlatformNegotiation.cs">
```

Comment:
Would it be possibly to move `PlatformNegotiation` into Microsoft.Build.Framework assembly and do not use Shared for it. 
In long term we would like to get rid of this Shared files shenanigans.